doc/prelimsLinux.rst

```diff
@@ -0,0 +1,222 @@
+Preliminary Steps for Linux Distributions
+=========================================
+
+1) Install Python 3.6+
+----------------------
+Check your current python version(s) by entering the following commands into the command line: 
+
+.. code-block:: shell
+
+   python --version
+   python3 --version
+   python3.6 --version
+
+If Python 3.6+ is not present, follow these instructions depending on your operating system:
+
+Running a **Linux** distribution, Python must be compiled. Download and compile the most recent Python release with the following commands (find the link to the correct .tar file `here <https://www.python.org/downloads/>`__):
+
+.. note::
+
+   If CMake is not already on your machine see step 3a to install it.
+
+.. code-block:: shell
+   
+   wget https://www.python.org/ftp/python/3.6.1/Python-3.6.1.tar.xz
+   sudo tar xzf Python-3.6.1.tar.xz
+   cd Python-3.6.1
+   ./configure       # sudo might be required
+   make altinstall   # sudo might be required
+
+Some **newer Linux** distributions may be able to install Python 3.6+ using apt-get.
+
+.. code-block:: shell
+
+   apt-get install python3.6   # sudo might be required
+
+.. note::
+
+   It can be helpful to use an alias to aid in differentiation between Python releases. paste the following lines into your ~/.bashrc or ~/.bash_aliases file (be sure to enter `source ~/.bashrc` or restart your shell):
+
+   .. code-block:: shell
+      
+      alias python=python3.6
+      alias pip=pip3.6
+
+
+2) Install virtualenvwrapper
+----------------------------
+
+Install and setup virtualenvwrapper and associated tools. This allows you to create isolated "virtual environments" with independent installations of Python packages. This isn't strictly necessary, but helps prevent conflicts between incompatible package versions. To install virtualenvwrapper, run the command below:
+
+.. code-block:: shell
+   
+   pip3.6 install virtualenvwrapper
+
+An introduction and walkthrough to using the virtualenvwrapper tool is found `here <https://virtualenvwrapper.readthedocs.io/en/latest/>`__. To use virtual environments, place the following lines in the .bashrc file. The first line sets the Python interpreter for your virtual environments to python3.6. The last line is a path to your shell startup file, change it depending on where virtualenvwrapper was installed by pip.
+
+.. code-block:: shell
+
+   VIRTUALENVWRAPPER_PYTHON=/usr/local/bin/python3.6
+   export WORKON_HOME=$HOME/.virtualenvs
+   export MSYS_HOME=/c/msys/1.0
```

Comment:
I added these to the .bashrc following the "Basic Installation" instructions on [this page](http://virtualenvwrapper.readthedocs.io/en/latest/install.html). They may not be necessary but my virtual environments are working with them included.